src/mongoose/helpers.ts

```diff
@@ -0,0 +1,23 @@
+import * as _ from 'underscore';
+
+export function pushMetadata<T>(
+  key: any,
+  target: any,
```

Comment:
Let's swap key and target, then it will be in a nor natural order.  Then add a comment if possible, for example:
```

/**
 * Push a reflection metadata array to the target.
 * @param target xxxx
 * @param xxx
 * @return xxx
 */
pushMetadata<T>(target: any, key: string | symbol, ...metadata: T[]): T[] {
}
```